bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerRecoveryOp.java

```diff
@@ -112,6 +112,7 @@ public void readLastConfirmedDataComplete(int rc, RecoveryData data) {
                             synchronized (lh) {
                                 lh.lastAddPushed = lh.lastAddConfirmed = data.getLastAddConfirmed();
                                 lh.length = data.getLength();
+                                lh.pendingAddsSequenceHead = lh.lastAddConfirmed + 1;
```

Comment:
I would like to avoid this logic change; I won't comment on the +1 part for the rest of the code.